netbout-web/src/main/java/com/netbout/rest/MarkdownTxtmark.java

```diff
@@ -47,10 +47,18 @@
         // @checkstyle LineLengthCheck (1 line)
         "(?<!\\]\\()(?<!=\")(https?:\\/\\/[a-zA-Z0-9-._~:\\?#@!$&'*+,;=%\\/]+[a-zA-Z0-9-_~#@$&'*+=%\\/])(?![\\w.]*\\]\\()"
     );
+    /**
+     * Pattern to detect lines which should have a line break on the end.
+     */
+    private static final Pattern NEW_LINE = Pattern.compile(
+        "^ {0,3}\\S* ?$", Pattern.MULTILINE
```

Comment:
@dmzaytsev right, thanks for explanation. Can you put all what you've written above to Javadoc? This is concept that is worth documenting.

Another thing - string `   a a ` has less than two spaces at the end and less than four spaces at the beginning, but is not matched by this pattern. Can you explain?